src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/Maintainability/AvoidDeadConditionalCode_NullAnalysis.cs

```diff
@@ -7280,5 +7280,124 @@ private void Method3(IReadOnlyList<int>? items3 = null)
                 },
             }.RunAsync();
         }
+
+        [Trait(Traits.DataflowAnalysis, Traits.Dataflow.PointsToAnalysis)]
+        [Trait(Traits.DataflowAnalysis, Traits.Dataflow.NullAnalysis)]
+        [Trait(Traits.DataflowAnalysis, Traits.Dataflow.CopyAnalysis)]
+        [Fact, WorkItem(6520, "https://github.com/dotnet/roslyn-analyzers/issues/6520")]
+        public async Task CompareTwoUnrelatedEnumVariables_NoDiagnosticsAsync()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+public class Parser
+{
+    private static ParsedPredicate ParsePredicate(ParsedPredicate predicate, PredicateOperand identifier, PredicateOperand literal)
+    {
+        if (predicate.Left.TypePrimitive == PredicateTypePrimitive.F1)
+        {
+            identifier = predicate.Left;
+            literal = predicate.Right;
+        }
+        else
+        {
+            identifier = predicate.Right;
+            literal = predicate.Left;
+        }
+
+        if (identifier.TypePrimitive != PredicateTypePrimitive.F1)
```

Comment:
The core issue here was that we were correctly deducing that `identifer` and `literal` can point to the exact same set of two potential runtime objects stored in `predicate.Left` and `predicate.Right`. However, the value deduction from predicate in this line `identifier.TypePrimitive != PredicateTypePrimitive.F1` shouldn't be applied to `literal.TypePrimitive`, which was the case prior to this fix. Now, for cases where an entity's i. nstance location can point to more than one runtime location, we also store the entity for this instance location, i.e. entity for `identifier` in this case, in the analysis entity created for `identifier.TypePrimitive`. This allows us to avoid applying value deduction for `identifier.TypePrimitive` to `literal.TypePrimitive`.